### ISSUE TITLE
Need a option to have TCP timeouts

### ISSUE THREAD

**Issue by Power2All** [panjf2000/gnet#170]
**Is your feature request related to a problem? Please describe.**
I am looking for a fast and simple TCP/HTTP timeout function.
I've tried a couple of things to do this properly, but it seems epoll is giving me troubles when I try to close.
Also, TCP eats a ton of resources on GNET, how should I approach this properly ?

**Describe the solution you'd like**
A simple way (and non-blocking, super fast like GNET itself is) to put a timeout for a TCP connection.

**Describe alternatives you've considered**
Trying to track down all TCP connections like I did with UDP, but TCP is by itself already very strained with resource usage ?

**Additional context**
Basically, TCP timeout is a basic thing that should be there by default ?

If there is a example of a already existing, and fast concurrency way of applying a timeout, please do let me know !

**Comment by Power2All** [comment]
Here is the message I receive, or it seems it's trying to close the connection (using a timeout function), but is not able to, cause epoll says it's already gone and away...

```
2021-01-03T17:14:39.765+0100    WARN    netpoll/epoll.go:125    Error occurs in user-defined function, the fd=66 in event-loop(1) is already closed, skipping it
github.com/panjf2000/gnet/internal/netpoll.(*Poller).Polling
        /root/go/src/github.com/panjf2000/gnet/internal/netpoll/epoll.go:125
github.com/panjf2000/gnet.(*eventloop).loopRun
        /root/go/src/github.com/panjf2000/gnet/eventloop_unix.go:85
github.com/panjf2000/gnet.(*server).startEventLoops.func1.1
        /root/go/src/github.com/panjf2000/gnet/server_unix.go:78
```
